include/nbl/builtin/hlsl/sort/counting.hlsl

```diff
@@ -0,0 +1,137 @@
+// Copyright (C) 2018-2024 - DevSH Graphics Programming Sp. z O.O.
+// This file is part of the "Nabla Engine".
+// For conditions of distribution and use, see copyright notice in nabla.h
+
+#include "nbl/builtin/hlsl/sort/common.hlsl"
+#include "nbl/builtin/hlsl/workgroup/arithmetic.hlsl"
+
+#ifndef _NBL_BUILTIN_HLSL_SORT_COUNTING_INCLUDED_
+#define _NBL_BUILTIN_HLSL_SORT_COUNTING_INCLUDED_
+
+namespace nbl
+{
+namespace hlsl
+{
+namespace sort
+{
+
+template<
+    uint16_t GroupSize,
+    uint16_t KeyBucketCount,
+    typename Key,
+    typename KeyAccessor,
+    typename ValueAccessor,
+    typename ScratchAccessor,
```

Comment:
I'd rename `scratch` to `histogram` throughout the code

Reply:
I thought it was apparent when I wrote, but oh well: https://github.com/Devsh-Graphics-Programming/Nabla/pull/690#discussion_r1604074297

Reply:
Fixed